// @foo-software/binoculars/dist/config/groups/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var a11y_aria_1 = require("./a11y-aria");
Object.defineProperty(exports, "getA11yAria", { enumerable: true, get: function () { return a11y_aria_1.default; } });
var a11y_audio_video_1 = require("./a11y-audio-video");
Object.defineProperty(exports, "getA11yAudioVideo", { enumerable: true, get: function () { return a11y_audio_video_1.default; } });
var a11y_best_practices_1 = require("./a11y-best-practices");
Object.defineProperty(exports, "getA11yBestPractices", { enumerable: true, get: function () { return a11y_best_practices_1.default; } });
var a11y_color_contrast_1 = require("./a11y-color-contrast");
Object.defineProperty(exports, "getA11yColorContrast", { enumerable: true, get: function () { return a11y_color_contrast_1.default; } });
var a11y_language_1 = require("./a11y-language");
Object.defineProperty(exports, "getA11yLanguage", { enumerable: true, get: function () { return a11y_language_1.default; } });
var a11y_names_labels_1 = require("./a11y-names-labels");
Object.defineProperty(exports, "getA11yNamesLabels", { enumerable: true, get: function () { return a11y_names_labels_1.default; } });
var a11y_navigation_1 = require("./a11y-navigation");
Object.defineProperty(exports, "getA11yNavigation", { enumerable: true, get: function () { return a11y_navigation_1.default; } });
var a11y_tables_lists_1 = require("./a11y-tables-lists");
Object.defineProperty(exports, "getA11yTablesLists", { enumerable: true, get: function () { return a11y_tables_lists_1.default; } });
var seo_content_1 = require("./seo-content");
Object.defineProperty(exports, "getSeoContent", { enumerable: true, get: function () { return seo_content_1.default; } });
var seo_crawl_1 = require("./seo-crawl");
Object.defineProperty(exports, "getSeoCrawl", { enumerable: true, get: function () { return seo_crawl_1.default; } });
var seo_mobile_1 = require("./seo-mobile");
Object.defineProperty(exports, "getSeoMobile", { enumerable: true, get: function () { return seo_mobile_1.default; } });
