// @foo-software/binoculars/dist/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var binoculars_1 = require("./binoculars");
Object.defineProperty(exports, "default", { enumerable: true, get: function () { return binoculars_1.default; } });
var accesskeys_1 = require("./definitions/accesskeys");
Object.defineProperty(exports, "accesskeys", { enumerable: true, get: function () { return accesskeys_1.default; } });
var aria_allowed_attr_1 = require("./definitions/aria-allowed-attr");
Object.defineProperty(exports, "ariaAllowedAttr", { enumerable: true, get: function () { return aria_allowed_attr_1.default; } });
var aria_hidden_body_1 = require("./definitions/aria-hidden-body");
Object.defineProperty(exports, "ariaHiddenBody", { enumerable: true, get: function () { return aria_hidden_body_1.default; } });
var aria_hidden_focus_1 = require("./definitions/aria-hidden-focus");
Object.defineProperty(exports, "ariaHiddenFocus", { enumerable: true, get: function () { return aria_hidden_focus_1.default; } });
var aria_input_field_name_1 = require("./definitions/aria-input-field-name");
Object.defineProperty(exports, "ariaInputFieldName", { enumerable: true, get: function () { return aria_input_field_name_1.default; } });
var aria_required_attr_1 = require("./definitions/aria-required-attr");
Object.defineProperty(exports, "ariaRequiredAttr", { enumerable: true, get: function () { return aria_required_attr_1.default; } });
var aria_required_children_1 = require("./definitions/aria-required-children");
Object.defineProperty(exports, "ariaRequiredChildren", { enumerable: true, get: function () { return aria_required_children_1.default; } });
var aria_required_parent_1 = require("./definitions/aria-required-parent");
Object.defineProperty(exports, "ariaRequiredParent", { enumerable: true, get: function () { return aria_required_parent_1.default; } });
var aria_roles_1 = require("./definitions/aria-roles");
Object.defineProperty(exports, "ariaRoles", { enumerable: true, get: function () { return aria_roles_1.default; } });
var aria_toggle_field_name_1 = require("./definitions/aria-toggle-field-name");
Object.defineProperty(exports, "ariaToggleFieldName", { enumerable: true, get: function () { return aria_toggle_field_name_1.default; } });
var aria_valid_attr_value_1 = require("./definitions/aria-valid-attr-value");
Object.defineProperty(exports, "ariaValidAttrValue", { enumerable: true, get: function () { return aria_valid_attr_value_1.default; } });
var aria_valid_attr_1 = require("./definitions/aria-valid-attr");
Object.defineProperty(exports, "ariaValidAttr", { enumerable: true, get: function () { return aria_valid_attr_1.default; } });
var button_name_1 = require("./definitions/button-name");
Object.defineProperty(exports, "buttonName", { enumerable: true, get: function () { return button_name_1.default; } });
var bypass_1 = require("./definitions/bypass");
Object.defineProperty(exports, "bypass", { enumerable: true, get: function () { return bypass_1.default; } });
var canonical_1 = require("./definitions/canonical");
Object.defineProperty(exports, "canonical", { enumerable: true, get: function () { return canonical_1.default; } });
var color_contrast_1 = require("./definitions/color-contrast");
Object.defineProperty(exports, "colorContrast", { enumerable: true, get: function () { return color_contrast_1.default; } });
var custom_controls_labels_1 = require("./definitions/custom-controls-labels");
Object.defineProperty(exports, "customControlsLabels", { enumerable: true, get: function () { return custom_controls_labels_1.default; } });
var custom_control_roles_1 = require("./definitions/custom-control-roles");
Object.defineProperty(exports, "customControlRoles", { enumerable: true, get: function () { return custom_control_roles_1.default; } });
var definition_list_1 = require("./definitions/definition-list");
Object.defineProperty(exports, "definitionList", { enumerable: true, get: function () { return definition_list_1.default; } });
var dlitem_1 = require("./definitions/dlitem");
Object.defineProperty(exports, "dlitem", { enumerable: true, get: function () { return dlitem_1.default; } });
var document_title_1 = require("./definitions/document-title");
Object.defineProperty(exports, "documentTitle", { enumerable: true, get: function () { return document_title_1.default; } });
var duplicate_id_active_1 = require("./definitions/duplicate-id-active");
Object.defineProperty(exports, "duplicateIdActive", { enumerable: true, get: function () { return duplicate_id_active_1.default; } });
var duplicate_id_aria_1 = require("./definitions/duplicate-id-aria");
Object.defineProperty(exports, "duplicateIdAria", { enumerable: true, get: function () { return duplicate_id_aria_1.default; } });
var focus_traps_1 = require("./definitions/focus-traps");
Object.defineProperty(exports, "focusTraps", { enumerable: true, get: function () { return focus_traps_1.default; } });
var focusable_controls_1 = require("./definitions/focusable-controls");
Object.defineProperty(exports, "focusableControls", { enumerable: true, get: function () { return focusable_controls_1.default; } });
var font_size_1 = require("./definitions/font-size");
Object.defineProperty(exports, "fontSize", { enumerable: true, get: function () { return font_size_1.default; } });
var form_field_multiple_labels_1 = require("./definitions/form-field-multiple-labels");
Object.defineProperty(exports, "formFieldMultipleLabels", { enumerable: true, get: function () { return form_field_multiple_labels_1.default; } });
var frame_title_1 = require("./definitions/frame-title");
Object.defineProperty(exports, "frameTitle", { enumerable: true, get: function () { return frame_title_1.default; } });
var heading_order_1 = require("./definitions/heading-order");
Object.defineProperty(exports, "headingOrder", { enumerable: true, get: function () { return heading_order_1.default; } });
var hreflang_1 = require("./definitions/hreflang");
Object.defineProperty(exports, "hreflang", { enumerable: true, get: function () { return hreflang_1.default; } });
var html_has_lang_1 = require("./definitions/html-has-lang");
Object.defineProperty(exports, "htmlHasLang", { enumerable: true, get: function () { return html_has_lang_1.default; } });
var html_lang_valid_1 = require("./definitions/html-lang-valid");
Object.defineProperty(exports, "htmlLangValid", { enumerable: true, get: function () { return html_lang_valid_1.default; } });
var http_status_code_1 = require("./definitions/http-status-code");
Object.defineProperty(exports, "httpStatusCode", { enumerable: true, get: function () { return http_status_code_1.default; } });
var image_alt_1 = require("./definitions/image-alt");
Object.defineProperty(exports, "imageAlt", { enumerable: true, get: function () { return image_alt_1.default; } });
var input_image_alt_1 = require("./definitions/input-image-alt");
Object.defineProperty(exports, "inputImageAlt", { enumerable: true, get: function () { return input_image_alt_1.default; } });
var interactive_element_affordance_1 = require("./definitions/interactive-element-affordance");
Object.defineProperty(exports, "interactiveElementAffordance", { enumerable: true, get: function () { return interactive_element_affordance_1.default; } });
var is_crawable_1 = require("./definitions/is-crawable");
Object.defineProperty(exports, "isCrawable", { enumerable: true, get: function () { return is_crawable_1.default; } });
var label_1 = require("./definitions/label");
Object.defineProperty(exports, "label", { enumerable: true, get: function () { return label_1.default; } });
var layout_table_1 = require("./definitions/layout-table");
Object.defineProperty(exports, "layoutTable", { enumerable: true, get: function () { return layout_table_1.default; } });
var link_name_1 = require("./definitions/link-name");
Object.defineProperty(exports, "linkName", { enumerable: true, get: function () { return link_name_1.default; } });
var link_text_1 = require("./definitions/link-text");
Object.defineProperty(exports, "linkText", { enumerable: true, get: function () { return link_text_1.default; } });
var list_1 = require("./definitions/list");
Object.defineProperty(exports, "list", { enumerable: true, get: function () { return list_1.default; } });
var listitem_1 = require("./definitions/listitem");
Object.defineProperty(exports, "listitem", { enumerable: true, get: function () { return listitem_1.default; } });
var logical_tab_order_1 = require("./definitions/logical-tab-order");
Object.defineProperty(exports, "logicalTabOrder", { enumerable: true, get: function () { return logical_tab_order_1.default; } });
var managed_focus_1 = require("./definitions/managed-focus");
Object.defineProperty(exports, "managedFocus", { enumerable: true, get: function () { return managed_focus_1.default; } });
var meta_description_1 = require("./definitions/meta-description");
Object.defineProperty(exports, "metaDescription", { enumerable: true, get: function () { return meta_description_1.default; } });
var meta_refresh_1 = require("./definitions/meta-refresh");
Object.defineProperty(exports, "metaRefresh", { enumerable: true, get: function () { return meta_refresh_1.default; } });
var meta_viewport_1 = require("./definitions/meta-viewport");
Object.defineProperty(exports, "metaViewport", { enumerable: true, get: function () { return meta_viewport_1.default; } });
var object_alt_1 = require("./definitions/object-alt");
Object.defineProperty(exports, "objectAlt", { enumerable: true, get: function () { return object_alt_1.default; } });
var offscreen_content_hidden_1 = require("./definitions/offscreen-content-hidden");
Object.defineProperty(exports, "offscreenContentHidden", { enumerable: true, get: function () { return offscreen_content_hidden_1.default; } });
var plugins_1 = require("./definitions/plugins");
Object.defineProperty(exports, "plugins", { enumerable: true, get: function () { return plugins_1.default; } });
var robots_txt_1 = require("./definitions/robots-txt");
Object.defineProperty(exports, "robotsTxt", { enumerable: true, get: function () { return robots_txt_1.default; } });
var structured_data_1 = require("./definitions/structured-data");
Object.defineProperty(exports, "structuredData", { enumerable: true, get: function () { return structured_data_1.default; } });
var tabindex_1 = require("./definitions/tabindex");
Object.defineProperty(exports, "tabindex", { enumerable: true, get: function () { return tabindex_1.default; } });
var tap_targets_1 = require("./definitions/tap-targets");
Object.defineProperty(exports, "tapTargets", { enumerable: true, get: function () { return tap_targets_1.default; } });
var td_headers_attr_1 = require("./definitions/td-headers-attr");
Object.defineProperty(exports, "tdHeadersAttr", { enumerable: true, get: function () { return td_headers_attr_1.default; } });
var th_has_data_cells_1 = require("./definitions/th-has-data-cells");
Object.defineProperty(exports, "thHasDataCells", { enumerable: true, get: function () { return th_has_data_cells_1.default; } });
var use_landmarks_1 = require("./definitions/use-landmarks");
Object.defineProperty(exports, "useLandmarks", { enumerable: true, get: function () { return use_landmarks_1.default; } });
var valid_lang_1 = require("./definitions/valid-lang");
Object.defineProperty(exports, "validLang", { enumerable: true, get: function () { return valid_lang_1.default; } });
var video_caption_1 = require("./definitions/video-caption");
Object.defineProperty(exports, "videoCaption", { enumerable: true, get: function () { return video_caption_1.default; } });
var video_description_1 = require("./definitions/video-description");
Object.defineProperty(exports, "videoDescription", { enumerable: true, get: function () { return video_description_1.default; } });
var viewport_1 = require("./definitions/viewport");
Object.defineProperty(exports, "viewport", { enumerable: true, get: function () { return viewport_1.default; } });
var visual_order_follows_dom_1 = require("./definitions/visual-order-follows-dom");
Object.defineProperty(exports, "visualOrderFollowsDom", { enumerable: true, get: function () { return visual_order_follows_dom_1.default; } });
