// @foo-software/binoculars/dist/config/lighthouseConfig.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const auditRefsConfig_1 = require("./auditRefsConfig");
const a11y_aria_1 = __importDefault(require("./groups/a11y-aria"));
const a11y_audio_video_1 = __importDefault(require("./groups/a11y-audio-video"));
const a11y_best_practices_1 = __importDefault(require("./groups/a11y-best-practices"));
const a11y_color_contrast_1 = __importDefault(require("./groups/a11y-color-contrast"));
const a11y_language_1 = __importDefault(require("./groups/a11y-language"));
const a11y_names_labels_1 = __importDefault(require("./groups/a11y-names-labels"));
const a11y_navigation_1 = __importDefault(require("./groups/a11y-navigation"));
const a11y_tables_lists_1 = __importDefault(require("./groups/a11y-tables-lists"));
const description_length_1 = __importDefault(require("../audits/seo/description-length"));
const elements_1 = __importDefault(require("../gatherers/elements"));
const headings_1 = __importDefault(require("../audits/seo/headings"));
const keywords_1 = __importDefault(require("../audits/seo/keywords"));
const meaningful_content_1 = __importDefault(require("./groups/meaningful-content"));
const meaningful_tag_structure_1 = __importDefault(require("../audits/seo/meaningful-tag-structure"));
const meaningful_text_1 = __importDefault(require("../audits/seo/meaningful-text"));
const seo_1 = __importDefault(require("./categories/seo"));
const seo_content_1 = __importDefault(require("./groups/seo-content"));
const seo_crawl_1 = __importDefault(require("./groups/seo-crawl"));
const seo_mobile_1 = __importDefault(require("./groups/seo-mobile"));
const title_element_1 = __importDefault(require("../gatherers/title-element"));
const title_length_1 = __importDefault(require("../audits/seo/title-length"));
const DEVTOOLS_RTT_ADJUSTMENT_FACTOR = 3.75;
const DEVTOOLS_THROUGHPUT_ADJUSTMENT_FACTOR = 0.9;
const throttling = {
    DEVTOOLS_RTT_ADJUSTMENT_FACTOR,
    DEVTOOLS_THROUGHPUT_ADJUSTMENT_FACTOR,
    mobileSlow4G: {
        rttMs: 150,
        throughputKbps: 1.6 * 1024,
        requestLatencyMs: 150 * DEVTOOLS_RTT_ADJUSTMENT_FACTOR,
        downloadThroughputKbps: 1.6 * 1024 * DEVTOOLS_THROUGHPUT_ADJUSTMENT_FACTOR,
        uploadThroughputKbps: 750 * DEVTOOLS_THROUGHPUT_ADJUSTMENT_FACTOR,
        cpuSlowdownMultiplier: 4,
    },
    mobileRegluar3G: {
        rttMs: 300,
        throughputKbps: 700,
        requestLatencyMs: 300 * DEVTOOLS_RTT_ADJUSTMENT_FACTOR,
        downloadThroughputKbps: 700 * DEVTOOLS_THROUGHPUT_ADJUSTMENT_FACTOR,
        uploadThroughputKbps: 700 * DEVTOOLS_THROUGHPUT_ADJUSTMENT_FACTOR,
        cpuSlowdownMultiplier: 4,
    },
    desktopDense4G: {
        rttMs: 40,
        throughputKbps: 10 * 1024,
        cpuSlowdownMultiplier: 1,
        requestLatencyMs: 0,
        downloadThroughputKbps: 0,
        uploadThroughputKbps: 0,
    },
};
exports.default = (locale) => ({
    extends: 'lighthouse:default',
    audits: [
        description_length_1.default,
        headings_1.default,
        keywords_1.default,
        meaningful_tag_structure_1.default,
        meaningful_text_1.default,
        title_length_1.default,
    ],
    categories: {
        binocularsSeo: {
            ...seo_1.default(locale),
            auditRefs: auditRefsConfig_1.seoAuditRefs,
        },
    },
    groups: {
        'a11y-aria': a11y_aria_1.default(locale),
        'a11y-audio-video': a11y_audio_video_1.default(locale),
        'a11y-best-practices': a11y_best_practices_1.default(locale),
        'a11y-color-contrast': a11y_color_contrast_1.default(locale),
        'a11y-language': a11y_language_1.default(locale),
        'a11y-names-labels': a11y_names_labels_1.default(locale),
        'a11y-navigation': a11y_navigation_1.default(locale),
        'a11y-tables-lists': a11y_tables_lists_1.default(locale),
        'meaningful-content': meaningful_content_1.default(),
        'seo-content': seo_content_1.default(locale),
        'seo-crawl': seo_crawl_1.default(locale),
        'seo-mobile': seo_mobile_1.default(locale),
    },
    passes: [
        {
            passName: 'defaultPass',
            gatherers: [elements_1.default, title_element_1.default],
        },
    ],
    settings: {
        emulatedFormFactor: 'mobile',
        onlyCategories: ['binocularsSeo'],
        throttling: throttling.mobileSlow4G,
        throttlingMethod: 'simulate',
    },
});
