// @foo-software/binoculars/dist/audits/seo/headings.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UIStrings = void 0;
const lighthouse_1 = require("lighthouse");
exports.UIStrings = {
    title: 'Has `<h1>` and `<h2>` headings',
    failureTitle: 'Does not have both `<h1>` and `<h2>` headings',
    description: 'Headings are important in signaling a hierarchy of content to the search engine.',
};
class Headings extends lighthouse_1.Audit {
    static get meta() {
        return {
            id: 'headings',
            title: exports.UIStrings.title,
            failureTitle: exports.UIStrings.failureTitle,
            description: exports.UIStrings.description,
            requiredArtifacts: ['Elements'],
        };
    }
    static audit(artifacts) {
        var _a, _b;
        const elements = artifacts.Elements;
        return {
            score: (((_a = elements === null || elements === void 0 ? void 0 : elements.h1) === null || _a === void 0 ? void 0 : _a.length) && ((_b = elements === null || elements === void 0 ? void 0 : elements.h2) === null || _b === void 0 ? void 0 : _b.length) && 1) || 0,
        };
    }
}
exports.default = Headings;
