// @foo-software/binoculars/dist/audits/seo/meaningful-text.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UIStrings = exports.MIN_TEXT_LENGTH = void 0;
const lighthouse_1 = require("lighthouse");
exports.MIN_TEXT_LENGTH = 300;
exports.UIStrings = {
    title: 'Has sufficient textual content (300 characters)',
    failureTitle: `Document body does not have at least ${exports.MIN_TEXT_LENGTH} characters`,
    description: `A minimal amount of textual content is necessary for search engines to pickup. Document body should have at least ${exports.MIN_TEXT_LENGTH} characters.`,
};
class MeaningfulText extends lighthouse_1.Audit {
    static get meta() {
        return {
            id: 'meaningful-text',
            title: exports.UIStrings.title,
            failureTitle: exports.UIStrings.failureTitle,
            description: exports.UIStrings.description,
            requiredArtifacts: ['Elements'],
        };
    }
    static audit(artifacts) {
        var _a;
        const [body] = ((_a = artifacts.Elements) === null || _a === void 0 ? void 0 : _a.body) || [];
        return {
            score: !body || !body.text || body.text.length < exports.MIN_TEXT_LENGTH ? 0 : 1,
        };
    }
}
exports.default = MeaningfulText;
