// @foo-software/binoculars/dist/audits/seo/meaningful-tag-structure.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UIStrings = void 0;
const lighthouse_1 = require("lighthouse");
const MIN_DIVERSITY = 2;
exports.UIStrings = {
    title: `Has at least ${MIN_DIVERSITY} different informational HTML tags of the following types: \`<p>\`, \`<li>\`, \`<img>\`, \`<table>\``,
    failureTitle: `Does not have at least ${MIN_DIVERSITY} different informational HTML tags of the following types: \`<p>\`, \`<li>\`, \`<img>\`, \`<table>\``,
    description: `Semantic HTML with a variety of tags defining content will improve signaling to search engines. Should have at least ${MIN_DIVERSITY} different informational HTML tags of the following types: \`<p>\`, \`<li>\`, \`<img>\`, \`<table>\``,
};
class MeaningfulTagStructure extends lighthouse_1.Audit {
    static get meta() {
        return {
            id: 'meaningful-tag-structure',
            title: exports.UIStrings.title,
            failureTitle: exports.UIStrings.failureTitle,
            description: exports.UIStrings.description,
            requiredArtifacts: ['Elements'],
        };
    }
    static audit(artifacts) {
        const { img, li, p, table } = artifacts.Elements || {};
        const diversity = [img, li, p, table].reduce((accumulator, current) => {
            if (!(current === null || current === void 0 ? void 0 : current.length)) {
                return accumulator;
            }
            return accumulator + 1;
        }, 0);
        return {
            score: diversity < MIN_DIVERSITY ? 0 : 1,
        };
    }
}
exports.default = MeaningfulTagStructure;
