// @foo-software/binoculars/dist/audits/seo/title-length.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UIStrings = exports.MIN_TEXT_LENGTH = exports.MAX_TEXT_LENGTH = void 0;
const lighthouse_1 = require("lighthouse");
exports.MAX_TEXT_LENGTH = 70;
exports.MIN_TEXT_LENGTH = 50;
exports.UIStrings = {
    title: 'Title length is appropriate',
    failureTitle: `Title is not between ${exports.MIN_TEXT_LENGTH} - ${exports.MAX_TEXT_LENGTH} characters`,
    description: `A title defines a descriptive summary of a page and should have the right amount of content for search engines to parse. Title should be between ${exports.MIN_TEXT_LENGTH} - ${exports.MAX_TEXT_LENGTH} characters.`,
    explanation: 'Title text is missing or empty.',
};
class TitleLength extends lighthouse_1.Audit {
    static get meta() {
        return {
            id: 'title-length',
            title: exports.UIStrings.title,
            failureTitle: exports.UIStrings.failureTitle,
            description: exports.UIStrings.description,
            requiredArtifacts: ['TitleElement'],
        };
    }
    static audit(artifacts) {
        const title = artifacts.TitleElement;
        if (!title || !title.text) {
            return {
                explanation: exports.UIStrings.explanation,
                score: 0,
            };
        }
        const trimmedTitleLength = title.text.trim().length;
        if (trimmedTitleLength < exports.MIN_TEXT_LENGTH ||
            trimmedTitleLength > exports.MAX_TEXT_LENGTH) {
            return {
                score: 0,
            };
        }
        return {
            score: 1,
        };
    }
}
exports.default = TitleLength;
