// @foo-software/binoculars/dist/config/audits/lighthouse/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var accesskeys_1 = require("./accesskeys");
Object.defineProperty(exports, "getAccesskeys", { enumerable: true, get: function () { return accesskeys_1.default; } });
var aria_allowed_attr_1 = require("./aria-allowed-attr");
Object.defineProperty(exports, "getAriaAllowedAttr", { enumerable: true, get: function () { return aria_allowed_attr_1.default; } });
var aria_hidden_body_1 = require("./aria-hidden-body");
Object.defineProperty(exports, "getAriaHiddenBody", { enumerable: true, get: function () { return aria_hidden_body_1.default; } });
var aria_hidden_focus_1 = require("./aria-hidden-focus");
Object.defineProperty(exports, "getAriaHiddenFocus", { enumerable: true, get: function () { return aria_hidden_focus_1.default; } });
var aria_input_field_name_1 = require("./aria-input-field-name");
Object.defineProperty(exports, "getAriaInputFieldName", { enumerable: true, get: function () { return aria_input_field_name_1.default; } });
var aria_required_attr_1 = require("./aria-required-attr");
Object.defineProperty(exports, "getAriaRequiredAttr", { enumerable: true, get: function () { return aria_required_attr_1.default; } });
var aria_required_children_1 = require("./aria-required-children");
Object.defineProperty(exports, "getAriaRequiredChildren", { enumerable: true, get: function () { return aria_required_children_1.default; } });
var aria_required_parent_1 = require("./aria-required-parent");
Object.defineProperty(exports, "getAriaRequiredParent", { enumerable: true, get: function () { return aria_required_parent_1.default; } });
var aria_roles_1 = require("./aria-roles");
Object.defineProperty(exports, "getAriaRoles", { enumerable: true, get: function () { return aria_roles_1.default; } });
var aria_toggle_field_name_1 = require("./aria-toggle-field-name");
Object.defineProperty(exports, "getAriaToggleFieldName", { enumerable: true, get: function () { return aria_toggle_field_name_1.default; } });
var aria_valid_attr_value_1 = require("./aria-valid-attr-value");
Object.defineProperty(exports, "getAriaValidAttrValue", { enumerable: true, get: function () { return aria_valid_attr_value_1.default; } });
var aria_valid_attr_1 = require("./aria-valid-attr");
Object.defineProperty(exports, "getAriaValidAttr", { enumerable: true, get: function () { return aria_valid_attr_1.default; } });
var button_name_1 = require("./button-name");
Object.defineProperty(exports, "getButtonName", { enumerable: true, get: function () { return button_name_1.default; } });
var bypass_1 = require("./bypass");
Object.defineProperty(exports, "getBypass", { enumerable: true, get: function () { return bypass_1.default; } });
var canonical_1 = require("./canonical");
Object.defineProperty(exports, "getCanonical", { enumerable: true, get: function () { return canonical_1.default; } });
var color_contrast_1 = require("./color-contrast");
Object.defineProperty(exports, "getColorContrast", { enumerable: true, get: function () { return color_contrast_1.default; } });
var crawlable_anchors_1 = require("./crawlable-anchors");
Object.defineProperty(exports, "getCrawlableAnchors", { enumerable: true, get: function () { return crawlable_anchors_1.default; } });
var custom_controls_labels_1 = require("./custom-controls-labels");
Object.defineProperty(exports, "getCustomControlsLabels", { enumerable: true, get: function () { return custom_controls_labels_1.default; } });
var custom_controls_roles_1 = require("./custom-controls-roles");
Object.defineProperty(exports, "getCustomControlsRoles", { enumerable: true, get: function () { return custom_controls_roles_1.default; } });
var definition_list_1 = require("./definition-list");
Object.defineProperty(exports, "getDefinitionList", { enumerable: true, get: function () { return definition_list_1.default; } });
var dlitem_1 = require("./dlitem");
Object.defineProperty(exports, "getDlitem", { enumerable: true, get: function () { return dlitem_1.default; } });
var document_title_1 = require("./document-title");
Object.defineProperty(exports, "getDocumentTitle", { enumerable: true, get: function () { return document_title_1.default; } });
var duplicate_id_active_1 = require("./duplicate-id-active");
Object.defineProperty(exports, "getDuplicateIdActive", { enumerable: true, get: function () { return duplicate_id_active_1.default; } });
var duplicate_id_aria_1 = require("./duplicate-id-aria");
Object.defineProperty(exports, "getDuplicateIdAria", { enumerable: true, get: function () { return duplicate_id_aria_1.default; } });
var focus_traps_1 = require("./focus-traps");
Object.defineProperty(exports, "getFocusTraps", { enumerable: true, get: function () { return focus_traps_1.default; } });
var focusable_controls_1 = require("./focusable-controls");
Object.defineProperty(exports, "getFocusableControls", { enumerable: true, get: function () { return focusable_controls_1.default; } });
var font_size_1 = require("./font-size");
Object.defineProperty(exports, "getFontSize", { enumerable: true, get: function () { return font_size_1.default; } });
var form_field_multiple_labels_1 = require("./form-field-multiple-labels");
Object.defineProperty(exports, "getFormFieldMultipleLabels", { enumerable: true, get: function () { return form_field_multiple_labels_1.default; } });
var frame_title_1 = require("./frame-title");
Object.defineProperty(exports, "getFrameTitle", { enumerable: true, get: function () { return frame_title_1.default; } });
var heading_order_1 = require("./heading-order");
Object.defineProperty(exports, "getHeadingOrder", { enumerable: true, get: function () { return heading_order_1.default; } });
var hreflang_1 = require("./hreflang");
Object.defineProperty(exports, "getHreflang", { enumerable: true, get: function () { return hreflang_1.default; } });
var html_has_lang_1 = require("./html-has-lang");
Object.defineProperty(exports, "getHtmlHasLang", { enumerable: true, get: function () { return html_has_lang_1.default; } });
var html_lang_valid_1 = require("./html-lang-valid");
Object.defineProperty(exports, "getHtmlLangValid", { enumerable: true, get: function () { return html_lang_valid_1.default; } });
var http_status_code_1 = require("./http-status-code");
Object.defineProperty(exports, "getHttpStatusCode", { enumerable: true, get: function () { return http_status_code_1.default; } });
var image_alt_1 = require("./image-alt");
Object.defineProperty(exports, "getImageAlt", { enumerable: true, get: function () { return image_alt_1.default; } });
var input_image_alt_1 = require("./input-image-alt");
Object.defineProperty(exports, "getInputImageAlt", { enumerable: true, get: function () { return input_image_alt_1.default; } });
var interactive_element_affordance_1 = require("./interactive-element-affordance");
Object.defineProperty(exports, "getInteractiveElementAffordance", { enumerable: true, get: function () { return interactive_element_affordance_1.default; } });
var is_crawlable_1 = require("./is-crawlable");
Object.defineProperty(exports, "getIsCrawlable", { enumerable: true, get: function () { return is_crawlable_1.default; } });
var label_1 = require("./label");
Object.defineProperty(exports, "getLabel", { enumerable: true, get: function () { return label_1.default; } });
var layout_table_1 = require("./layout-table");
Object.defineProperty(exports, "getLayoutTable", { enumerable: true, get: function () { return layout_table_1.default; } });
var link_name_1 = require("./link-name");
Object.defineProperty(exports, "getLinkName", { enumerable: true, get: function () { return link_name_1.default; } });
var link_text_1 = require("./link-text");
Object.defineProperty(exports, "getLinkText", { enumerable: true, get: function () { return link_text_1.default; } });
var list_1 = require("./list");
Object.defineProperty(exports, "getList", { enumerable: true, get: function () { return list_1.default; } });
var listitem_1 = require("./listitem");
Object.defineProperty(exports, "getListitem", { enumerable: true, get: function () { return listitem_1.default; } });
var logical_tab_order_1 = require("./logical-tab-order");
Object.defineProperty(exports, "getLogicalTabOrder", { enumerable: true, get: function () { return logical_tab_order_1.default; } });
var managed_focus_1 = require("./managed-focus");
Object.defineProperty(exports, "getManagedFocus", { enumerable: true, get: function () { return managed_focus_1.default; } });
var meta_description_1 = require("./meta-description");
Object.defineProperty(exports, "getMetaDescription", { enumerable: true, get: function () { return meta_description_1.default; } });
var meta_refresh_1 = require("./meta-refresh");
Object.defineProperty(exports, "getMetaRefresh", { enumerable: true, get: function () { return meta_refresh_1.default; } });
var meta_viewport_1 = require("./meta-viewport");
Object.defineProperty(exports, "getMetaViewport", { enumerable: true, get: function () { return meta_viewport_1.default; } });
var object_alt_1 = require("./object-alt");
Object.defineProperty(exports, "getObjectAlt", { enumerable: true, get: function () { return object_alt_1.default; } });
var offscreen_content_hidden_1 = require("./offscreen-content-hidden");
Object.defineProperty(exports, "getOffscreenContentHidden", { enumerable: true, get: function () { return offscreen_content_hidden_1.default; } });
var plugins_1 = require("./plugins");
Object.defineProperty(exports, "getPlugins", { enumerable: true, get: function () { return plugins_1.default; } });
var robots_txt_1 = require("./robots-txt");
Object.defineProperty(exports, "getRobotsTxt", { enumerable: true, get: function () { return robots_txt_1.default; } });
var structured_data_1 = require("./structured-data");
Object.defineProperty(exports, "getStructuredData", { enumerable: true, get: function () { return structured_data_1.default; } });
var tabindex_1 = require("./tabindex");
Object.defineProperty(exports, "getTabindex", { enumerable: true, get: function () { return tabindex_1.default; } });
var tap_targets_1 = require("./tap-targets");
Object.defineProperty(exports, "getTapTargets", { enumerable: true, get: function () { return tap_targets_1.default; } });
var td_headers_attr_1 = require("./td-headers-attr");
Object.defineProperty(exports, "getTdHeadersAttr", { enumerable: true, get: function () { return td_headers_attr_1.default; } });
var th_has_data_cells_1 = require("./th-has-data-cells");
Object.defineProperty(exports, "getThHasDataCells", { enumerable: true, get: function () { return th_has_data_cells_1.default; } });
var use_landmarks_1 = require("./use-landmarks");
Object.defineProperty(exports, "getUseLandmarks", { enumerable: true, get: function () { return use_landmarks_1.default; } });
var valid_lang_1 = require("./valid-lang");
Object.defineProperty(exports, "getValidLang", { enumerable: true, get: function () { return valid_lang_1.default; } });
var video_caption_1 = require("./video-caption");
Object.defineProperty(exports, "getVideoCaption", { enumerable: true, get: function () { return video_caption_1.default; } });
var video_description_1 = require("./video-description");
Object.defineProperty(exports, "getVideoDescription", { enumerable: true, get: function () { return video_description_1.default; } });
var viewport_1 = require("./viewport");
Object.defineProperty(exports, "getViewport", { enumerable: true, get: function () { return viewport_1.default; } });
var visual_order_follows_dom_1 = require("./visual-order-follows-dom");
Object.defineProperty(exports, "getVisualOrderFollowsDom", { enumerable: true, get: function () { return visual_order_follows_dom_1.default; } });
